test/versioning/maven.spec.js

```diff
@@ -70,3 +77,56 @@ describe('versioning/maven/compare', () => {
     expect(compare('1', '1-cr1')).toEqual(1);
   });
 });
+
+describe('versioning/maven/index', () => {
+  it('validates version string', () => {
+    expect(isVersion('')).toEqual(false);
+    expect(isVersion('v1.0.0')).toEqual(false);
+    expect(isVersion('1.0.0')).toEqual(true);
+    expect(isVersion('0')).toEqual(true);
+    expect(isVersion('0.1-2-sp')).toEqual(true);
+    expect(isVersion('1-final')).toEqual(true);
+  });
+  it('checks if version is stable', () => {
+    expect(isStable('')).toEqual(null);
+    expect(isStable('foobar')).toEqual(null);
+    expect(isStable('v1.0.0')).toEqual(null);
+    expect(isStable('1')).toEqual(true);
+    expect(isStable('1.2')).toEqual(true);
+    expect(isStable('1.2.3')).toEqual(true);
+    expect(isStable('1.2.3.4')).toEqual(true);
+    expect(isStable('1-alpha-1')).toEqual(false);
+    expect(isStable('1-b1')).toEqual(false);
+    expect(isStable('1.final')).toEqual(true);
+    expect(isStable('1.0milestone1')).toEqual(false);
+    expect(isStable('1-sp')).toEqual(true);
+    expect(isStable('1-ga-1')).toEqual(true);
+  });
+  it('returns major version', () => {
+    expect(getMajor('')).toEqual(null);
+    expect(getMajor('v1.0.0')).toEqual(null);
+    expect(getMajor('1')).toEqual(1);
+    expect(getMajor('1.2')).toEqual(1);
+    expect(getMajor('1.2.3')).toEqual(1);
+    expect(getMajor('1rc42')).toEqual(1);
+  });
+  it('returns minor version', () => {
+    expect(getMinor('')).toEqual(null);
+    expect(getMinor('v1.0.0')).toEqual(null);
+    expect(getMinor('1')).toEqual(0);
```

Comment:
I am not sure if we should be returning 0 or null here. 0 is probably OK for now